### PR TITLE
Configure logrotate

### PR DESCRIPTION
**Pull request by elemoine**
This configures logrotate for `/var/log/collectd.log` and `/var/log/remote_collectd.log`.

### CODE REVIEW

File: collectd/files/collectd_logrotate.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/var/log/{{ service_name }}.log {
```
Comment (SwannCroiset) [inline collectd/files/collectd_logrotate.conf:1]: you may want to use these options:
copytruncate
compress         
delaycompress    

and optionally these ones as well:
minsize 20M  
maxsize 50M

File: collectd/_service.sls [hunk 2]
```diff
@@ -53,6 +53,16 @@
   - require_in:
     - file: {{ client.service }}_client_conf_dir_clean
 
+/etc/logrotate.d/{{ client.service }}:
```
Comment (SwannCroiset) [inline collectd/_service.sls:56]: I'd add a dependency 'require' on  {{ client.config_dir }}/00_collectd_logfile.conf:

**Review by SwannCroiset — CHANGES_REQUESTED** [review]

File: collectd/_service.sls (hunk 2 above)
Comment (elemoine) [reply collectd/_service.sls:56]: As discussed this is not necessary.

File: collectd/files/collectd_logrotate.conf (hunk 1 above)
Comment (elemoine) [reply collectd/files/collectd_logrotate.conf:1]: Thanks. I checked and `copytruncate` is not necessary. Actually the collectd logfile plugin automatically recreates the `/var/log/collectd.log` file if it was removed. So we can even use the `nocreate` option and let the collectd logfile plugin create the file by itself (with the proper mode, user, etc.).

**Comment by elemoine** [comment]
Patch updated and tested. Please review again.

File: collectd/files/collectd_logrotate.conf [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  rotate 5
+  daily
+  nocreate
+  missingok
+  compress
+  delaycompress
+  minsize 20M
+  maxsize 50M
+  su root syslog
```
Comment (SwannCroiset) [inline collectd/files/collectd_logrotate.conf]: I think that "su root syslog" is herited from the main logrotate configuration file, then is not mandatory here

Comment (elemoine) [reply collectd/files/collectd_logrotate.conf]: That's what I thought too. But that did not work as expected. I'll try again.

Comment (elemoine) [reply collectd/files/collectd_logrotate.conf]: I was testing using `logrotate /etc/logrotate.d/collectd` and the inheritance mechanism does not work in that case.

I removed the `su` option as you suggested.

**Review by SwannCroiset — APPROVED** [review]

**Pull request merged**